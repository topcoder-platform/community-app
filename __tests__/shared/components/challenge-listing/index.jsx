import React from 'react';
import _ from 'lodash';
import Renderer from 'react-test-renderer/shallow';

let ChallengeListing = require('components/challenge-listing').default;

const selectBucket = jest.fn();
const setFilterState = jest.fn();
const setSort = jest.fn();

const mockData1 = {
  activeBucket: 'abc',
  challenges: [],
  myChallenges: [],
  filterState: {},
  lastUpdateOfActiveChallenges: 1500124537142,
  loadingChallenges: false,
  loadingDraftChallenges: false,
  loadingPastChallenges: false,
  selectBucket,
  setFilterState,
  setSort,
  sorts: {},
  auth: {},
};

const mockData2 = _.extend({}, mockData1, {
  communityFilter: {},
  loadingChallenges: true,
});

describe('Matches shallow shapshot 1', () => {
  test('shapshot 1', () => {
    const renderer = new Renderer();

    renderer.render((
      <ChallengeListing {...mockData1} />
    ));
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});


describe('Matches shallow shapshot 2', () => {
  beforeEach(() => {
    jest.resetModules();
    ChallengeListing = require('components/challenge-listing').default;
  });

  afterAll(() => {
    jest.clearAllMocks();
  });

  test('shapshot 2', () => {
    // require('utils/config');
    const renderer = new Renderer();

    renderer.render((
      <ChallengeListing {...mockData2} />
    ));
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
