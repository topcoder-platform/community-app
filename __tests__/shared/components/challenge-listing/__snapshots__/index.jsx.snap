// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches shallow shapshot 1 shapshot 1 1`] = `
<div
  className="src-shared-components-challenge-listing-___style__ChallengeFiltersExample___3IjeI"
  id="challengeFilterContainer"
>
  <Connect(Container)
    challenges={Array []}
    communityName={null}
    isAuth={false}
    isCardTypeSet="Challenges"
    setCardType={[Function]}
  />
  <div
    className="src-shared-components-challenge-listing-___style__tc-content-wrapper___1MqlF src-shared-components-challenge-listing-___style__hidden___4C3Up"
  >
    <div
      className="src-shared-components-challenge-listing-___style__sidebar-container-mobile___3e65d"
    />
    <div
      className="src-shared-components-challenge-listing-___style__challenges-container___1invl src-shared-components-challenge-listing-___style__SRMs-container___1Kmjf"
    >
      <div>
        <SRMCard
          category="now"
          srmChallenge={null}
        />
      </div>
      <div>
        <div
          className="src-shared-components-challenge-listing-___style__title___2KPNT"
        >
          Upcoming SRMs
        </div>
      </div>
      <div>
        <div
          className="src-shared-components-challenge-listing-___style__title___2KPNT"
        >
          Past SRMs
        </div>
        <SRMCard
          category="past"
          srmChallenge={null}
        />
      </div>
    </div>
    <div
      className="src-shared-components-challenge-listing-___style__sidebar-container-desktop___h0bz6"
    >
      <Sticky
        activeClass="active"
        bottomBoundary={0}
        enableTransforms={true}
        enabled={true}
        onStateChange={null}
        releasedClass="released"
        shouldFreeze={[Function]}
        top={20}
      />
    </div>
  </div>
  <div
    className="src-shared-components-challenge-listing-___style__tc-content-wrapper___1MqlF"
  >
    <div
      className="src-shared-components-challenge-listing-___style__sidebar-container-mobile___3e65d"
    >
      <Connect(SidebarContainer)
        loadMoreChallenges={null}
      />
    </div>
    <Connect(Listing)
      activeBucket="abc"
      auth={Object {}}
      challenges={Array []}
      communityName={null}
      expandTag={null}
      expandedTags={Array []}
      extraBucket={null}
      filterState={Object {}}
      loadMoreActive={null}
      loadMoreChallenges={null}
      loadMoreMy={null}
      loadMoreOnGoing={null}
      loadMoreOpen={null}
      loadMorePast={null}
      loadMoreReviewOpportunities={null}
      loadingActiveChallenges={false}
      loadingPastChallenges={false}
      newChallengeDetails={false}
      openChallengesInNewTabs={false}
      preListingMsg={null}
      prizeMode="money-usd"
      reviewOpportunities={Array []}
      selectBucket={[MockFunction]}
      setFilterState={[MockFunction]}
      setSort={[MockFunction]}
      sorts={Object {}}
    />
    <div
      className="src-shared-components-challenge-listing-___style__sidebar-container-desktop___h0bz6"
    >
      <Sticky
        activeClass="active"
        bottomBoundary="#challengeFilterContainer"
        enableTransforms={true}
        enabled={true}
        onStateChange={null}
        releasedClass="released"
        shouldFreeze={[Function]}
        top={20}
      >
        <Connect(SidebarContainer)
          extraBucket={null}
          hideTcLinksInFooter={false}
          loadMoreChallenges={null}
        />
      </Sticky>
    </div>
  </div>
</div>
`;

exports[`Matches shallow shapshot 2 shapshot 2 1`] = `
<div
  className="src-shared-components-challenge-listing-___style__ChallengeFiltersExample___3IjeI"
  id="challengeFilterContainer"
>
  <Connect(Container)
    challenges={Array []}
    communityName={null}
    isAuth={false}
    isCardTypeSet="Challenges"
    setCardType={[Function]}
  />
  <div
    className="src-shared-components-challenge-listing-___style__tc-content-wrapper___1MqlF src-shared-components-challenge-listing-___style__hidden___4C3Up"
  >
    <div
      className="src-shared-components-challenge-listing-___style__sidebar-container-mobile___3e65d"
    />
    <div
      className="src-shared-components-challenge-listing-___style__challenges-container___1invl src-shared-components-challenge-listing-___style__SRMs-container___1Kmjf"
    >
      <div>
        <SRMCard
          category="now"
          srmChallenge={null}
        />
      </div>
      <div>
        <div
          className="src-shared-components-challenge-listing-___style__title___2KPNT"
        >
          Upcoming SRMs
        </div>
      </div>
      <div>
        <div
          className="src-shared-components-challenge-listing-___style__title___2KPNT"
        >
          Past SRMs
        </div>
        <SRMCard
          category="past"
          srmChallenge={null}
        />
      </div>
    </div>
    <div
      className="src-shared-components-challenge-listing-___style__sidebar-container-desktop___h0bz6"
    >
      <Sticky
        activeClass="active"
        bottomBoundary={0}
        enableTransforms={true}
        enabled={true}
        onStateChange={null}
        releasedClass="released"
        shouldFreeze={[Function]}
        top={20}
      />
    </div>
  </div>
  <div
    className="src-shared-components-challenge-listing-___style__tc-content-wrapper___1MqlF"
  >
    <div
      className="src-shared-components-challenge-listing-___style__sidebar-container-mobile___3e65d"
    >
      <Connect(SidebarContainer)
        loadMoreChallenges={null}
      />
    </div>
    <div
      className="src-shared-components-challenge-listing-___style__challenge-cards-container___37tl8"
    >
      <div
        className="src-shared-components-challenge-listing-___style__ChallengeCardExamples___EHRPE"
      >
        <ChallengeCardPlaceholder
          id={0}
        />
        <ChallengeCardPlaceholder
          id={1}
        />
        <ChallengeCardPlaceholder
          id={2}
        />
        <ChallengeCardPlaceholder
          id={3}
        />
        <ChallengeCardPlaceholder
          id={4}
        />
        <ChallengeCardPlaceholder
          id={5}
        />
        <ChallengeCardPlaceholder
          id={6}
        />
        <ChallengeCardPlaceholder
          id={7}
        />
      </div>
    </div>
    <div
      className="src-shared-components-challenge-listing-___style__sidebar-container-desktop___h0bz6"
    >
      <Sticky
        activeClass="active"
        bottomBoundary="#challengeFilterContainer"
        enableTransforms={true}
        enabled={true}
        onStateChange={null}
        releasedClass="released"
        shouldFreeze={[Function]}
        top={20}
      >
        <Connect(SidebarContainer)
          extraBucket={null}
          hideTcLinksInFooter={false}
          loadMoreChallenges={null}
        />
      </Sticky>
    </div>
  </div>
</div>
`;
